### PR TITLE
Add new certs (and a rake task to help)

### PR DESCRIPTION
**Pull request by mitchellhenke**
Adds a task to make it easier to get new certs from our root certs that aren't necessarily replacements for existing certs. I added all of the DoD ones, but I wasn't sure if we should add the `WidePoint ORC SSP 5` issued by Federal Common Policy CA

Output:
```
mitchellehenke:identity-pki/ (main*) $ rake certs:find_new
  Expiration: 2030-11-05 14:16:00 UTC
  Subject: /C=US/O=ORC PKI/CN=WidePoint ORC SSP 5
  Issuer: /C=US/O=U.S. Government/OU=FPKI/CN=Federal Common Policy CA G2
  SHA1 Fingerpint: 80f4731a60fd5f2eb0468d0629310daa50ad210d
  Key ID: 23:B8:4E:B1:4E:6D:24:44:8B:44:67:A7:65:CF:A1:3B:39:94:66:DC
  In Certificate Store: false
Would you like to save this cert? Type yes to save.
no
  Expiration: 2027-06-02 14:11:23 UTC
  Subject: /C=US/O=U.S. Government/OU=DoD/OU=PKI/CN=DOD ID CA-65
  Issuer: /C=US/O=U.S. Government/OU=DoD/OU=PKI/CN=DoD Root CA 3
  SHA1 Fingerpint: 2838d25ae351654a094f00348f4bd0ea3178d871
  Key ID: 62:E0:48:38:56:6D:6F:6B:25:BE:BB:C3:88:58:B1:AC:9C:43:C9:5B
  In Certificate Store: false
Would you like to save this cert? Type yes to save.
yes
Writing certificate to ./config/certs/C=US, O=U.S. Government, OU=DoD, OU=PKI, CN=DOD ID CA-65.pem
  Expiration: 2027-06-09 13:55:26 UTC
  Subject: /C=US/O=U.S. Government/OU=DoD/OU=PKI/CN=DOD EMAIL CA-65
  Issuer: /C=US/O=U.S. Government/OU=DoD/OU=PKI/CN=DoD Root CA 3
  SHA1 Fingerpint: 671288d3adbb5909aa2858e3f86498ded6fd85a0
  Key ID: 98:B6:34:0A:1B:ED:04:9A:53:0A:8A:05:73:FA:42:67:CD:10:66:B6
  In Certificate Store: false
Would you like to save this cert? Type yes to save.
yes
Writing certificate to ./config/certs/C=US, O=U.S. Government, OU=DoD, OU=PKI, CN=DOD EMAIL CA-65.pem
  Expiration: 2027-06-02 14:05:19 UTC
  Subject: /C=US/O=U.S. Government/OU=DoD/OU=PKI/CN=DOD EMAIL CA-64
  Issuer: /C=US/O=U.S. Government/OU=DoD/OU=PKI/CN=DoD Root CA 3
  SHA1 Fingerpint: 8ca4fcf4d1186f52e243be7b8cccfeb0ec7d4f4e
  Key ID: 8B:90:7D:21:EB:BC:1A:15:7B:C2:B0:4E:7E:73:51:C8:0E:B6:DC:B6
  In Certificate Store: false
Would you like to save this cert? Type yes to save.
yes
Writing certificate to ./config/certs/C=US, O=U.S. Government, OU=DoD, OU=PKI, CN=DOD EMAIL CA-64.pem
  Expiration: 2027-06-09 13:57:18 UTC
  Subject: /C=US/O=U.S. Government/OU=DoD/OU=PKI/CN=DOD SW CA-66
  Issuer: /C=US/O=U.S. Government/OU=DoD/OU=PKI/CN=DoD Root CA 3
  SHA1 Fingerpint: 8f9d91c33d4b4e4e6fd7690c053048a7aabbd3a2
  Key ID: EB:06:53:98:30:86:9E:DF:5E:8B:0B:A1:26:26:FA:A6:61:0F:B9:94
  In Certificate Store: false
Would you like to save this cert? Type yes to save.
yes
Writing certificate to ./config/certs/C=US, O=U.S. Government, OU=DoD, OU=PKI, CN=DOD SW CA-66.pem
  Expiration: 2027-06-09 13:58:25 UTC
  Subject: /C=US/O=U.S. Government/OU=DoD/OU=PKI/CN=DOD SW CA-67
  Issuer: /C=US/O=U.S. Government/OU=DoD/OU=PKI/CN=DoD Root CA 3
  SHA1 Fingerpint: 7b38aa22d6f76a8ff48b23d2485e7d2520f99cab
  Key ID: DB:B4:AB:A3:8C:29:45:BD:AD:2A:56:37:CC:54:5D:BC:D7:FD:2A:27
  In Certificate Store: false
Would you like to save this cert? Type yes to save.
yes
Writing certificate to ./config/certs/C=US, O=U.S. Government, OU=DoD, OU=PKI, CN=DOD SW CA-67.pem
  Expiration: 2027-06-09 13:51:38 UTC
  Subject: /C=US/O=U.S. Government/OU=DoD/OU=PKI/CN=DOD EMAIL CA-62
  Issuer: /C=US/O=U.S. Government/OU=DoD/OU=PKI/CN=DoD Root CA 3
  SHA1 Fingerpint: cc04a4f733b767761de8935d4c745eb25524b505
  Key ID: CD:C5:E6:E3:E4:25:17:97:70:BA:AA:93:E2:B9:BE:41:A3:92:2B:E7
  In Certificate Store: false
Would you like to save this cert? Type yes to save.
yes
Writing certificate to ./config/certs/C=US, O=U.S. Government, OU=DoD, OU=PKI, CN=DOD EMAIL CA-62.pem
  Expiration: 2027-06-02 14:02:21 UTC
  Subject: /C=US/O=U.S. Government/OU=DoD/OU=PKI/CN=DOD EMAIL CA-63
  Issuer: /C=US/O=U.S. Government/OU=DoD/OU=PKI/CN=DoD Root CA 3
  SHA1 Fingerpint: 1b977e3104f27cd4afb47d502e09037a956ab126
  Key ID: 4D:31:AD:51:D6:4E:57:7E:67:69:33:25:03:7E:C6:29:A5:DD:BA:F3
  In Certificate Store: false
Would you like to save this cert? Type yes to save.
yes
Writing certificate to ./config/certs/C=US, O=U.S. Government, OU=DoD, OU=PKI, CN=DOD EMAIL CA-63.pem
  Expiration: 2027-06-02 14:07:31 UTC
  Subject: /C=US/O=U.S. Government/OU=DoD/OU=PKI/CN=DOD ID CA-62
  Issuer: /C=US/O=U.S. Government/OU=DoD/OU=PKI/CN=DoD Root CA 3
  SHA1 Fingerpint: 14f4cfd8364412a6a27e5bba82c5342ff9b337a7
  Key ID: 6E:7B:20:04:4D:11:C9:81:E9:16:35:90:D4:CB:C2:5B:42:D6:07:58
  In Certificate Store: false
Would you like to save this cert? Type yes to save.
yes
Writing certificate to ./config/certs/C=US, O=U.S. Government, OU=DoD, OU=PKI, CN=DOD ID CA-62.pem
  Expiration: 2027-06-02 14:09:37 UTC
  Subject: /C=US/O=U.S. Government/OU=DoD/OU=PKI/CN=DOD ID CA-64
  Issuer: /C=US/O=U.S. Government/OU=DoD/OU=PKI/CN=DoD Root CA 3
  SHA1 Fingerpint: d9991bd1e89ae5a8b1143c3c37f01103779b8db7
  Key ID: 57:BC:C0:5E:DB:89:AB:3B:DD:47:A9:72:1E:AE:CB:42:19:6B:1E:7A
  In Certificate Store: false
Would you like to save this cert? Type yes to save.
yes
Writing certificate to ./config/certs/C=US, O=U.S. Government, OU=DoD, OU=PKI, CN=DOD ID CA-64.pem
```

### CODE REVIEW

File: lib/tasks/certs.rake [hunk 1]
```diff
@@ -117,4 +117,44 @@ namespace :certs do
...
+  # modified to go deeper in the issuing tree.
+  desc 'Find certs issued by existing certs'
+  task find_new: :environment do
+    root_keys = IdentityConfig.store.trusted_ca_root_identifiers + IdentityConfig.store.dod_root_identifiers
+    root_keys.each do |key_id|
+      root_cert = CertificateStore.instance[key_id]
+      repository_certs = IssuingCaService.fetch_ca_repository_certs_for_cert(root_cert)
+
+      repository_certs.each do |repo_cert|
+        next if CertificateStore.instance.certificates.any? { |x| x.serial == repo_cert.serial }
```
Comment (zachmargolis) [inline lib/tasks/certs.rake:135]: Do we need to load the certificate store after each cert is written?

File: lib/tasks/certs.rake [hunk 2]
```diff
@@ -117,4 +117,44 @@ namespace :certs do
...
+        puts "  Expiration: #{repo_cert.not_after}"
+        puts "  Subject: #{repo_cert.subject}"
+        puts "  Issuer: #{repo_cert.issuer}"
+        puts "  SHA1 Fingerpint: #{repo_cert.sha1_fingerprint}"
+        puts "  Key ID: #{repo_cert.key_id}"
+        puts "  In Certificate Store: #{CertificateStore.instance[repo_cert.key_id].present?}"
+        puts "Would you like to save this cert? Type yes to save."
+        input = STDIN.gets.strip
+
+        if input == 'yes'
```
Comment (zachmargolis) [inline lib/tasks/certs.rake:146]: how about also accepting `y` because we are lazy people

**Review by zachmargolis — APPROVED** [review]
LGTM! glad to finally have a version of this script

File: lib/tasks/certs.rake (hunk 1 above)
Comment (mitchellhenke) [reply lib/tasks/certs.rake:135]: We _should_ be able to trust PKI stuff to not collide, but added in 084a09e for safety

File: lib/tasks/certs.rake (hunk 2 above)
Comment (mitchellhenke) [reply lib/tasks/certs.rake]: good call, added in 084a09e

**Review by jmhooper — APPROVED** [review]

**Pull request merged**